### ISSUE TITLE
Rule S5604 Using intrusive permissions is security-sensitive

### ISSUE THREAD

**Issue by eric-therond-sonarsource** [SonarSource/SonarJS#1769]
Implements https://jira.sonarsource.com/browse/RSPEC-5604

For both JS/TS

Code examples: https://github.com/SonarSource/security-expected-issues/tree/master/javascript/rules/hotspots/RSPEC-5604

For the detection pattern see the comments in https://github.com/SonarSource/security-expected-issues/blob/master/javascript/rules/hotspots/RSPEC-5604/permissions.js

* Check if following syntax is used with `name` property set to the value listed in `permissions` rule parameter
```js
navigator.permissions.query({name:"geolocation"})
```

* Then for each permission from `permissions` parameter check possible APIs to use it (specified in https://github.com/SonarSource/security-expected-issues/blob/master/javascript/hotspots/RSPEC-5604/permissions.js)

* Message should depend on the reported permission/feature. e.g. `Make sure the use of the geolocation is necessary. ` if issue is reported on geolocation feature usage.
  * Camera and microphone are requested both at the same time when `getUserMedia()` method is called so in this case the message should be  `Make sure the use of the camera and microphone is necessary. `
